### PR TITLE
Initial cut at restructuring of APIs

### PR DESCRIPTION
**Pull request by rtyler**
See the commit messages for more specific details

### CODE REVIEW

File: features/api/artifact/reading.feature [hunk 1]
```diff
@@ -22,15 +22,28 @@ Feature: Artifact READ APIs
...
+
+    It is encouraged to specify the version of the API you would like to
+    receive inside of a Accept header. If this header is not specified than the
+    client is expected to receive the v1 version of the JSON API
+
+    Given there is an artifact
+    When I GET "/api/artifacts/1" with custom headers:
+      | Header Name | Value                            |
+      | Accept      | application/vnd.deploydb.v1+json |
+    Then the response should be 200
```
Comment (GiriDandu) [inline features/api/artifact/reading.feature]: Can we add a test error case for invalid media type, to make sure we return 406

Comment (rtyler) [inline features/api/artifact/reading.feature]: Totally, technically that's what this scenario does right now whether I want it to or not though  :(

File: src/cucumber/groovy/step_definitions/ArtifactSteps.groovy [hunk 2]
```diff
@@ -21,3 +21,25 @@ Given(~/^there are artifacts$/) { ->
...
+    name  = var == "name" ? randomString : "cukes"
+    version = var == "version" ? randomString : "1.2.345"
+    sourceUrl = var == "sourceUrl" ? randomString : "http://example.com/cucumber.jar"
+
+    requestBody = """ {
+        "group" : "$group",
+        "name" : "$name",
+        "version" : "$version",
+        "sourceUrl" : "$sourceUrl"
+      }
```
Comment (GiriDandu) [inline src/cucumber/groovy/step_definitions/ArtifactSteps.groovy]: Not sure why this is moved here.

Comment (rtyler) [inline src/cucumber/groovy/step_definitions/ArtifactSteps.groovy]: While I was updating some steps I noticed that this was in the wrong file (HttpSteps) which was not the right place for this to live

**Pull request merged**